lib/proxy-agent.js

```diff
@@ -82,40 +82,38 @@ class ProxyAgent extends DispatcherBase {
     this[kClient] = clientFactory(resolvedUrl, { connect })
     this[kAgent] = new Agent({
       ...opts,
-      connect: async (opts, callback) => {
+      connect: (opts, callback) => {
```

Comment:
this was unnecessary and made it less maintainable...

Reply:
This one was also questionable for me. But  it should save a Promise?